### ISSUE TITLE
using windows pipes as ports

### ISSUE THREAD

**Issue by gurpreetatwal** [mozilla/node-convict#171]
When deploying node apps on azure websites, azure exposes the port that the app should listen on as the `PORT` environment variable. This can cause issues when using convict, as the port validator expects ports to only be numbers. The simplest solution would be to use another variable for the production config but that produces a weird situation where the port your app listens on is a different variable in production vs development. 

This could be fixed by modifying the current port validator or adding a new validator that accepts a pipe or a port.

Here's a draft I've written: 
```js
convict.addFormat({
  name: 'pipeport',
  validate: function(val) {
    val = String(val).includes('\\\\.\\pipe\\') ? val : parseInt(val);
    if (typeof val === 'number' && (isNaN(val) || val < 0 || val > 65535)) {
      throw new RangeError('Ports must be within range [0 - 65535]');
    }
  },
 coerce: function(val) {
    return String(val).includes('\\\\.\\pipe\\') ? val : parseInt(val);
  },
});
```

Thoughts on adding such a validator or modifying the current port validator to accept pipes?

[msdn doc on pipes](https://msdn.microsoft.com/en-us/library/windows/desktop/aa365783%28v=vs.85%29.aspx)
[blog post about azure ports](http://www.geekwithopinions.com/2013/12/09/node-js-and-azure-is-it-a-port-or-a-pipe/)

**Comment by madarche** [comment]
Hello @gurpreetatwal. I had never heard about azure ports before. Thanks for the description!

Your custom format seems to be a good solution since it allows both this special azure port string and the classical integer form.

Alternatively you can also accept anything for the `port` property:
```javascript
port: {
    doc: "The port to bind.",
    format: "*",
    default: 8080,
    env: "PORT"
  }
```

**Comment by gurpreetatwal** [comment]
haha I hadn't either until I used Azure.

Ah okay I had never tried using `"*"` as the format, removing the format entirely but leaving the default was causing convict to infer the format from the default value and was coercing the pipe into a `NaN`.  Took quite some time to pinpoint that as the issue. 

Regarding the fix, do you think it's too specific to add as a default format or would it be fine? If you choose to not add I definitely think it should be documented somewhere. 

As an aside meteorjs added support for pipes: meteor/meteor#4413

**Comment by madarche** [comment]
I agree to add a new `named_pipe` predefined format if you want to contribute it.

As for the idea of adding a new `named_pipe_or_port`, why not? Waiting for your PR.

**Comment by gurpreetatwal** [comment]
Just wanted to give an update, I've written the new formats however I still have to write tests for them. I'll make PR once I'm good on my side

**Comment by madarche** [comment]
Fix by #175
